File: MosPosudit.UI/shared/lib/dtos/payment/paypal_capture_response.dart
class PayPalCaptureResponse {
  final String orderId;
  final String transactionId;
  final String status;
  final num amount;
  final bool isCompleted;
  final int databasePaymentId;

  PayPalCaptureResponse({
    required this.orderId,
    required this.transactionId,
    required this.status,
    required this.amount,
    required this.isCompleted,
    required this.databasePaymentId,
  });

  factory PayPalCaptureResponse.fromJson(Map<String, dynamic> json) =>
      PayPalCaptureResponse(
        orderId: json['orderId'] ?? json['OrderId'] ?? '',
        transactionId: json['transactionId'] ?? json['TransactionId'] ?? '',
        status: json['status'] ?? json['Status'] ?? '',
        amount: json['amount'] ?? json['Amount'] ?? 0,
        isCompleted: json['isCompleted'] ?? json['IsCompleted'] ?? false,
        databasePaymentId: json['databasePaymentId'] ?? json['DatabasePaymentId'] ?? 0,
      );
}

File: MosPosudit.Model/Responses/Payment/PayPalCaptureResponse.cs
namespace MosPosudit.Model.Responses.Payment
{
    public class PayPalCaptureResponse
    {
        public string OrderId { get; set; } = string.Empty;
        public string TransactionId { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public bool IsCompleted { get; set; }
        public int DatabaseOrderId { get; set; }
        public int DatabasePaymentId { get; set; }
    }
}

